src/stores/room.ts

```diff
@@ -0,0 +1,82 @@
+import { getDetailRoomApi, createRoomApi, settingRoomApi } from '@/services/room'
+import type { IRoom } from '@/types/room'
+import { showToast } from '@/utils/toast'
+import { defineStore } from 'pinia'
+import router from '@/routers/router'
+import { apiError } from '@/utils/exceptionHandler'
+
+export const useRoomStore = defineStore({
+  id: 'room',
+  state: () => ({
+    isLoading: false,
+    detailRoom: {} as IRoom,
+    currentSetting: {
+      quizzfly_id: '',
+      is_show_question: false,
+      is_auto_play: false,
+      lobby_music: 'string',
+    },
+  }),
+  actions: {
+    async initRoom(val: any) {
+      try {
+        this.isLoading = true
+        const { data } = await createRoomApi(val)
+        this.setDetailRoom(data)
+        router.push({
+          name: 'lobby',
+        })
+      } catch (error) {
+        console.error(error)
+        showToast({
+          description: apiError(error).message,
+          variant: 'destructive',
+        })
+      }
+      setTimeout(() => {
```

Comment:
` setTimeout(() => {
        this.isLoading = true
      }, 5000)
      this.isLoading = false
`
sửa thành
`
 setTimeout(() => {
        this.isLoading = false
      }, 5000)
     `